pkg/object/httpserver/mux.go

```diff
@@ -431,3 +467,40 @@ func (mi *muxInstance) exportPrometheusMetrics(stat *httpstat.Metric, backend st
 	mi.metrics.RequestSizeBytesPercentage.With(labels).Observe(float64(stat.ReqSize))
 	mi.metrics.ResponseSizeBytesPercentage.With(labels).Observe(float64(stat.RespSize))
 }
+
+func newAccessLogFormatter(format string) *accessLogFormatter {
+	if format == "" {
+		format = defaultAccessLogFormat
+	}
+	escapeReg := regexp.MustCompile(`(\{|\}|\[|\])`)
+	expr := escapeReg.ReplaceAllString(format, "{{`$1`}}")
+	varReg := regexp.MustCompile(`\$([a-zA-z]*)`)
+	expr = varReg.ReplaceAllString(expr, "{{.$1}}")
+	tpl := template.Must(template.New("").Parse(expr))
+	return &accessLogFormatter{template: tpl}
+}
+
+func (formatter *accessLogFormatter) format(log *accessLog) string {
+	var buf bytes.Buffer
+	if err := formatter.template.Execute(&buf, log); err != nil {
+		logger.Errorf("format access log failed: %v", err)
+	}
+	return buf.String()
+}
+
+func printHeader(header http.Header) string {
+	buf := bytes.Buffer{}
+	i := 0
+	for key, values := range header {
+		if i > 0 {
+			buf.WriteString(", ")
+		}
+		buf.WriteString(fmt.Sprintf("%v: %v", key, values))
+		i++
+	}
+	return buf.String()
+}
+
+func printSize(size uint64) string {
+	return fmt.Sprintf("%vB", size)
```

Comment:
we can add `B` to the format string and avoid using `fmt.Sprintf` (and `printSize`) because `fmt.Sprintf` is slow.

Reply:
Ok, I see the problem. Seems we need to find a better way for the format string...

Reply:
yeah, `fmt. Sprintf` is a bit slow, so I replace it to `strconv.FormatUint() + "B"` which is faster.